src/librustc_typeck/check/method/suggest.rs

```diff
@@ -78,6 +78,30 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             return;
         }
 
+        let print_disambiguation_help = |
+            err: &mut DiagnosticBuilder<'_>,
+            trait_name: String,
+        | {
+            err.help(&format!("to disambiguate the method call, write `{}::{}({}{})` \
+                              instead",
+                              trait_name,
+                              item_name,
+                              if rcvr_ty.is_region_ptr() && args.is_some() {
+                                  if rcvr_ty.is_mutable_pointer() {
+                                      "&mut "
+                                  } else {
+                                      "&"
+                                  }
+                              } else {
+                                  ""
+                              },
+                              args.map(|arg| arg.iter()
+                                  .map(|arg| print::to_string(print::NO_ANN,
+                                                              |s| s.print_expr(arg)))
```

Comment:
It'd be nice to use `span_to_snippet` to keep the current formatting instead of relying on `print::to_string` first.